module/zfs/dbuf.c

```diff
@@ -952,10 +952,8 @@ dbuf_verify(dmu_buf_impl_t *db)
 		ASSERT3U(db->db.db_offset, ==, db->db_blkid * db->db.db_size);
 	}
 
-	for (dr = db->db_data_pending; dr != NULL; dr = dr->dr_next)
-		ASSERT(dr->dr_dbuf == db);
-
-	for (dr = db->db_last_dirty; dr != NULL; dr = dr->dr_next)
+	for (dr = list_head(&db->db_dirty_records);
+	    dr != NULL; dr = list_next(&db->db_dirty_records, dr))
 		ASSERT(dr->dr_dbuf == db);
```

Comment:
Since we're already traversing the entire list to verify `dr->dr_dbuf` let's take the opportunity to ASSERT that it's in `txg` order from highest to lowest.